### PR TITLE
[WIP] CI: Use SQL Server Developer Edition

### LINKED ISSUE #15122: Use separate SQL Server machine for C.I. tests
Currently by far our main source of test flakiness is caused by SQL Server failing to respond to connections or otherwise flaking out in similar ways. One idea to help make this better is to run SQL Server on a different box when using the C.I. such that:
* It's not competing for resources with all the tests running
* It's using a real TCP/IP connection, rather than a local connection (e.g. named pipes) to LocalDb

### PR DESCRIPTION
**Pull request by bricelam**
Fixes #15122

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -93,6 +86,13 @@ jobs:
         - job: Linux
           pool:
             vmImage: ubuntu-16.04
+          container:
```
Comment (smitpatel) [inline azure-pipelines.yml]: yay! Linux testing!

**Comment by bricelam** [comment]
Lame. Our Windows pool doesn't have Docker.

**Comment by bricelam** [comment]
☹️
> SqlException : A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: TCP Provider, error: 40 - Could not open a connection to SQL Server)

**Comment by bricelam** [comment]
Yep, Windows is a lost cause with this approach--we need new agents first.

**Comment by bricelam** [comment]
Well, I got it working on Linux... 😬

**Comment by ajcvickers** [comment]
@bricelam So is this a Windows container containing SQL Server, with the container running on Linux?

**Comment by bricelam** [comment]
No, Linux on Linux. Need to explore other options for Windows.

**Comment by bricelam** [comment]
I’ll send a new PR since this exploitation ended up pretty far away from the original goal. I also need to increase the timeout...

**Pull request closed without merging**